crates/proc-macro-srv/src/lib.rs

```diff
@@ -59,9 +63,16 @@ impl ProcMacroSrv {
 
         let macro_body = task.macro_body.to_subtree();
         let attributes = task.attributes.map(|it| it.to_subtree());
-        let result = expander
-            .expand(&task.macro_name, &macro_body, attributes.as_ref())
-            .map(|it| FlatTree::new(&it));
+        let result = crossbeam::scope(|s| {
+            s.spawn(|_| {
+                expander
+                    .expand(&task.macro_name, &macro_body, attributes.as_ref())
+                    .map(|it| FlatTree::new(&it))
+            })
+            .join()
+            .unwrap()
+        })
+        .unwrap();
```

Comment:
It would be nice if in both cases `std::panic::resume_unwind` is used to avoid having 3 backtraces when there is a panic. Otherwise users will likely end up reporting one of the other panics rather than the root panic.